### DISCUSSION TITLE
useUrlState hook

### DISCUSSION THREAD

**Discussion by karan316** [mantinedev/mantine#5907] [Feature requests]
A useUrlState hook would be really useful to manage url specific state like navigation, selected item, etc. 

Something similar to this - [useUrlState](https://ahooks.js.org/hooks/use-url-state). A drawback of this is that it requires the usage of react-router. Is there a way in which we can create a library agnostic version of the same hook?

**Comment by rtivital** [comment]
This logic is specific to routing library you use, it is not a good idea to include it in a components/hooks library.
